### PR TITLE
Feature improvement 3

### PR DESCRIPTION
**Pull request by deekshithmd**
Fixed few bugs, added search feature

### CODE REVIEW

File: src/components/Navigation/Navigation.jsx [hunk 1]
```diff
@@ -1,28 +1,37 @@
...
+  const debounce = (cb) => {
+    let timer;
+    return function (...args) {
+      clearTimeout(timer);
+      timer = setTimeout(() => cb.call(this, ...args), 300);
+    };
+  };
+
+  const process = debounce((query) => searchProducts(query));
+
```
Comment (anik31) [inline src/components/Navigation/Navigation.jsx:34]: appreciation - nice that you've used debounce for search

File: src/hooks/userActions.jsx [hunk 2]
```diff
@@ -97,6 +97,18 @@ export const useUserActions = () => {
...
+  //     filtered.price === 200
+  //       ? filtered.products
+  //       : filtered.products.filter(
+  //           (item) => parseInt(item.price) <= parseInt(filtered.price)
+  //         );
+  //   dispatch({ type: "FILTERED", payload: o });
+  // };
+
+  // const getCategoryFiltered=(filtered,)
+
```
Comment (anik31) [inline src/hooks/userActions.jsx]: suggestion - you can remove commented code if not required or else you can mention that it is for future work

File: src/pages/profile-page/Profile.jsx [hunk 3]
```diff
@@ -10,9 +10,7 @@ export default function Profile() {
   const { successToast } = useToast();
 
   const logoutHandler = () => {
-    localStorage.removeItem("login");
-    localStorage.removeItem("cart");
-    localStorage.removeItem("wishlist");
+    localStorage.clear();
     setToken(false);
     successToast("Succefully Logged Out...");
```
Comment (anik31) [inline src/pages/profile-page/Profile.jsx]: suggestion - you can correct the spelling inside toast

**Review by anik31 — APPROVED** [review]
Everything looks good, the search is working fine in the preview and code is also clean.

**Review by tushargoyal812 — COMMENTED** [review]
code is clean and to the point.
variable naming is good.
looks good to merge

**Pull request merged**